devclean.sh

```diff
@@ -37,19 +34,10 @@ settings () {
 cat << EOF_SETTINGS
 Settings:
 
-  SRW_DIR=${SRW_DIR}
-  BUILD_DIR=${BUILD_DIR}
-  BIN_DIR=${BIN_DIR}
```

Comment:
We need to have BIN_DIR option to have set up by the user. This is required for working with singularity container, when the binaries are built in a non-standard location. A standard location is used to have links to a container wrapper script.

Reply:
SRW_DIR needs to be set here, to make sure it is not empty, before this variable is used by a "rm" command. This addresses the concerns in the note [https://www.quora.com/What-are-some-crazy-rm-rf-stories-you-have-heard-about](https://www.quora.com/What-are-some-crazy-rm-rf-stories-you-have-heard-about) that was mentioned in https://github.com/ufs-community/ufs-srweather-app/issues/1073

Reply:
In what context would a user be cleaning/rebuilding in an existing Singularity container? This feels like a very niche use case that only an advanced user would be using.

If absolutely necessary, I could include a `--singularity` flag that would delete the location that is used in a singularity, but I don't understand the reason why this needs to be in a different directory, and why instructions can't just be to "remove this directory for a clean build". Also, this use case is also not covered in the users guide or documentation that I can find, so users wouldn't know how to do this anyway!

Again, I'll emphasize my arguments against including custom directories to feed into this script to be deleted: If a user wants to delete a directory in a custom location, they already need to know that location exists and must be deleted. In this case, it is extra work for them to provide flags to this clean script rather than directly deleting that directory that they know they need to remove! So including these custom directory flags is 1. extra work for the user, and 2. extra work for developers

Reply:
@mkavulich @MichaelLueken 
The current way the software container was set to work was to launch a wrapper script in place of the executable that starts singularity container first. That _requires_ binary directory to be different from a default, because the default directory, i.e., where the model expects to find executables, needs to contain a wrapper script for the executables that starts singularity container first. Sharing a presentation that is viewable for all NOAA accounts, and also for some additional accounts:
https://docs.google.com/presentation/d/1yMZG5b4FHTvzf4cwNSAjmbye9-LybWMoukLONw9eR_E/edit?usp=sharing

Slides 40,42 demonstrate the way this is approached.
The way the links in a standard binary location (./exec/) are created is a bit tricky and requires doing very precise steps. If model cleaning is required, these links created in ./exec/ directory need to remain; BIN_DIR with the actual binaries needs to be cleaned.

As for now, singularity container is _the only way_ we offer current SRW use for the public at this point and conduct trainings. Using spack-stack to build the libraries for the SRW is neither very accessible for the public nor could be easily built on users machines.